test/integration/authz_v2_test.go

```diff
@@ -0,0 +1,399 @@
+	switch a {
+	case create:
+		_, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Create(athenzDomain, metav1.CreateOptions{})
+		assert.Nil(t, err, "athenz domain create error should be nil")
+	case update:
+		currentAD, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Get(athenzDomain.Name, metav1.GetOptions{})
+		assert.Nil(t, err, "athenz domain get error should be nil")
+		athenzDomain.ResourceVersion = currentAD.ResourceVersion
+		_, err = framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Update(athenzDomain, metav1.UpdateOptions{})
+		assert.Nil(t, err, "athenz domain update error should be nil")
+	case delete:
+		err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Delete(athenzDomain.Name, metav1.DeleteOptions{})
+		assert.Nil(t, err, "athenz domain delete error should be nil")
+	}
+}
+
+// applyServices Based on the action specified creates, updates or deletes services
+func applyServices(t *testing.T, services []*v1.Service, a action) {
+	switch a {
+	case create:
+		createServiceObjects(t, services)
+	case update:
+		updateServiceObjects(t, services)
+	case delete:
+		deleteServiceObjects(t, services)
+	}
+}
+
+// validateAuthorizationPolicy - validates that the authorizationPolicies's spec generated is same as the one passed in
+func validateAuthorizationPolicy(t *testing.T, authorizationPolicies []*model.Config) {
+	err := wait.PollImmediate(time.Second, time.Second*5, func() (bool, error) {
+		for _, policy := range authorizationPolicies {
+			got := framework.Global.IstioClientset.Get(policy.GroupVersionKind(), policy.ConfigMeta.Name, policy.ConfigMeta.Namespace)
+			if got == nil {
+				return false, nil
+			}
+
+			if !reflect.DeepEqual(got.Spec, policy.Spec) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	assert.Nil(t, err, "Failed to validate AuthorizationPolicy")
+}
+
+// verifyAuthorizationPolicyIsRemoved - makes sure that AuthorizationPolicy with namespace is not present.
+func verifyAuthorizationPolicyIsRemoved(t *testing.T, authPolicy *model.Config) {
+	err := wait.PollImmediate(time.Second, time.Second*5, func() (bool, error) {
+		authPolicyList, err := framework.Global.IstioClientset.List(authPolicy.GroupVersionKind(), authPolicy.Namespace)
+		if err != nil {
+			return false, err
+		}
+		if len(authPolicyList) != 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	assert.Nil(t, err, "Authorization policy list not empty")
+}
+
+// Basic test case
+// 1. Athenz Domain exists, created service, checked for Authorization Policy
+// Initial: No services, existing AD
+// Input Actions: create service with annotation
+// Output: AP created
+func TestCreateAuthorizationPolicy(t *testing.T) {
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Manual update AP
```

Comment:
possible to add a case which override annotaion is set for an ap?

Reply:
Added testcase